### ISSUE TITLE
EXPORT NAME isidentifier() check prevents exporting into, e.g. AudioWorkletGlobalScope

### ISSUE THREAD

**Issue by olilarkin** [emscripten-core/emscripten#17319]
Forgive me if I can fix this easily in my code, but since a fairly recent emscripten version, I have problems exporting a wasm module which needs to be in the AudioWorkletGlobalScope. It seems like it is a valid js indentifier, e.g. **AudioWorkletGlobalScope.WAM.MyWebAudioModule**, but the emcc python script doesn't allow it. If I comment out this check it builds and works fine.

https://github.com/emscripten-core/emscripten/blob/4b5c4f0694174e081661944ab3ffdb43dd1949b8/emcc.py#L2572

What would be the best solution?

**Comment by olilarkin** [comment]
ps thanks for all the work on emscripten!

**Comment by sbc100** [comment]
`AudioWorkletGlobalScope.WAM.MyWebAudioModule` is really three different identifiers isn't it?

Can you do `-sEXPORT_NAME=MyWebAudioModule` and then do `AudioWorkletGlobalScope.WAM.MyWebAudioModule = MyWebAudioModule` in a `--post-js`?

**Comment by jariseon** [comment]
yeah the dots (due to namespacing) probably makes the python check unhappy. sbc100's solution looks good.

iirc WAM namespace was added to reduce global namespace pollution, but it is not strictly required as long as the export names are unique. nowadays there's also `globalThis` to replace the hacky `AudioWorkletGlobalScope` workaround. although that is unrelated to this specific issue.
